vcf_annotation_tools/vcf_info_annotator.py

```diff
@@ -32,13 +32,19 @@ def create_vcf_writer(args, vcf_reader):
     else:
         (head, sep, tail) = args.input_vcf.rpartition('.vcf')
         output_file = ('').join([head, '.info.vcf', tail])
+
     new_header = vcf_reader.header.copy()
-    od = OrderedDict([('ID', args.info_field), ('Number', '.'), ('Type', args.value_format), ('Description', args.description)])
-    if args.source:
-        od['Source'] = args.source
-    if args.version:
-        od['Version'] = args.version
-    new_header.add_info_line(od)
+
+    if args.w:
```

Comment:
I don't think this handles the case where the `-w` flag is set but the desired `info_field` doesn't exist yet in the header. If that is the case we should create the header. I believe you can achieve this by adding `and args.info_field in new_header.info_ids()` to this if-clause.

A test case for this situation would probably also be good.

Reply:
I think the right approach is to raise an exception if that happens - good call on finding that edge case and suggestion for the test. I'll add that